jdaviz/app.py

```diff
@@ -385,7 +385,8 @@ def get_data_from_viewer(self, viewer_reference, data_label=None,
                             layer_data = layer_data.get_object(cls=cls,
                                                                statistic=statistic)
                         else:
-                            layer_data = layer_data.get_object(cls=cls)
+                            layer_data = layer_data.get_object(cls=cls,
```

Comment:
It appears that now the if-else construct is not necessary, since both sides execute the same function call.

Reply:
Excellent point. I need to figure out what's causing this test failure though - maybe there was a good reason for the current state after all? WE still want consistency though, so I'll have to figure a way around the problem.

Reply:
Updated this to look for Spectrum1D as the target class in the if statement, rather than looking at the shape of the Data object, and only pass the statistic in that case.

Reply:
Your changes are exactly right. The `get_object` is just a proxy for the `to_object` method in the glue-astronomy translators. Only the `Spectrum1D` `to_object` has a `statistic` parameter. The assumption was made that 1D data would only use `Spectrum1D` objects (which may not be the case).